homeassistant/components/squeezebox/__init__.py

```diff
@@ -117,15 +128,54 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
     )
     _LOGGER.debug("LMS Device %s", device)
 
-    coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
+    server_coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
 
     entry.runtime_data = SqueezeboxData(
-        coordinator=coordinator,
+        coordinator=server_coordinator,
         server=lms,
     )
 
-    await coordinator.async_config_entry_first_refresh()
+    # set up player discovery
+    known_servers = hass.data.setdefault(DOMAIN, {}).setdefault(KNOWN_SERVERS, {})
+    known_players = known_servers.setdefault(lms.uuid, {}).setdefault(KNOWN_PLAYERS, [])
+
+    async def _player_discovery(now: datetime | None = None) -> None:
+        """Discover squeezebox players by polling server."""
+
+        async def _discovered_player(player: Player) -> None:
+            """Handle a (re)discovered player."""
+            if player.player_id in known_players:
+                async_dispatcher_send(
+                    hass, SIGNAL_PLAYER_REDISCOVERED, player.player_id, player.connected
+                )
+            else:
+                _LOGGER.debug("Adding new entity: %s", player)
+                player_coordinator = SqueezeBoxPlayerUpdateCoordinator(
+                    hass, player, lms.uuid
+                )
+                known_players.append(player.player_id)
+                async_dispatcher_send(
+                    hass, SIGNAL_PLAYER_DISCOVERED, player_coordinator
+                )
+
+        if players := await lms.async_get_players():
+            for player in players:
+                hass.async_create_task(_discovered_player(player))
+
+        entry.async_on_unload(
+            async_call_later(hass, DISCOVERY_INTERVAL, _player_discovery)
+        )
+
+    await server_coordinator.async_config_entry_first_refresh()
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    _LOGGER.debug(
+        "Adding player discovery job for LMS server: %s", entry.data[CONF_HOST]
+    )
+    entry.async_create_background_task(
+        hass, _player_discovery(), "squeezebox.media_player.player_discovery"
+    )
```

Comment:
What does this do? Because it's not executed regularly right?

Reply:
This queues up the first poll of the server for players. `_player_discovery()` looks for players, adds any new ones that it finds, and then sets up another poll after DISCOVERY_INTERVAL.